src/App.tsx

```diff
@@ -1,18 +1,64 @@
 import React from 'react';
 import './App.css';
+import { useSearchParams } from 'react-router-dom';
+import { Items } from './components/Items';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [searchParams, setSearchParams] = useSearchParams({});
+
+  const currentPage = searchParams.get('page') || '1';
+  const currentPerPage = searchParams.get('perPage') || '5';
+
+  const handleSelect = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = e.target;
+
+    searchParams.set('page', `${1}`);
+    searchParams.set('perPage', `${value}`);
+    setSearchParams(searchParams);
```

Comment:
seems like you mutate your state)
https://dev.to/ngprnk/how-to-use-usesearchparams-with-react-router-v6-4g5h
check this link